File: test_archives/py_tests/batch3/test_function7.py
from functions.function7 import function7
import pytest

def test_function7_1(): 
    assert function7("Hello", 2, [3, 4, 5]) == 22
    
def test_function7_2(): 
    assert function7("World", 3, [1,2,3,4,5]) == 30.0

File: test_archives/cpp_tests/batch1/test_function7.cpp
#include "../catch.hpp"
#include "../functions/function7.cpp"

TEST_CASE( "Testing function7 with test_function7_1" ) {
    REQUIRE( function7("Hello", 1, 2, true) == "Hello 3" );
}

TEST_CASE( "Testing function7 with test_function7_2" ) {
    REQUIRE( function7("World", 3, 4, false) == "World 12" );
}